lib/web_of_science/contributions.rb

```diff
@@ -34,28 +34,18 @@ def author_contribution(author, uid)
     # @param [Publication]
     # @return [Contribution]
     def find_or_create_contribution(author, publication)
-      contrib = publication.contributions.find_or_initialize_by(author_id: author.id)
-      create_contribution(author, contrib) unless contrib.persisted?
-      contrib
-    end
-
-    # Save a new contribution to a WOS-UID publication for author.
-    # @param author [Author]
-    # @param contrib [Contribution]
-    # @return [Boolean]
-    def create_contribution(author, contrib)
-      contrib.assign_attributes(
-        cap_profile_id: author.cap_profile_id,
-        featured: false, status: 'new', visibility: 'private'
-      )
-      contrib.save!
-      # Add the pub_hash[:authorship] data to the publication
-      contrib.publication.pubhash_needs_update!
-      contrib.publication.save!
+      publication.contributions.find_or_create_by!(author_id: author.id) do |contrib|
+        contrib.assign_attributes(
+          cap_profile_id: author.cap_profile_id,
+          featured: false, status: 'new', visibility: 'private'
+        )
+        # Add the pub_hash[:authorship] data to the publication
+        publication.pubhash_needs_update!
+        publication.save! # contrib.save! not needed
+      end
     rescue ActiveRecord::ActiveRecordError => err
-      message = "Failed to create contribution for author: #{author.id}, pub: #{contrib.publication.id}"
+      message = "Failed to find/create contribution for author: #{author.id}, pub: #{publication.id}"
       NotificationManager.error(err, message, self)
-      false
     end
```

Comment:
- guess this is OK, although not entirely clear about what happens with model callbacks that trigger changes to pub_hash data when using `create!` vs. `save!` - need to be sure that the callback behavior is the same.
- need to double check that `create_contribution` is not used anywhere